numpy/core/src/multiarray/item_selection.c

```diff
@@ -1749,6 +1750,281 @@ PyArray_LexSort(PyObject *sort_keys, int axis)
     return NULL;
 }
 
+/*
+* Helper function for PyArray_KeySort
+* Sorts 1D slice based on argsort indices
+* takes in raw bytes of data, indices array
+* Number of elements, and stride of data
+* element size and temporary storage for an element.
```

Comment:
Some style nitpicks, like a space before the `*`.   I like to try and do some doxygen style comments nowadays, but no need (style nitpicks also apply below, always space before and after the `*`).